js/src/launchAsync.ts

```diff
@@ -94,21 +96,23 @@ export function launchAsync(token: string, subdomain: string, content: Content,
             }
         };
 
+        const exit = (): void => {
+            reset();
+
+            // Execute exit callback if we have one
+            if (options.onExit) {
+                options.onExit();
+            }
+        }
+
         // Reset variables
         reset();
 
         const messageHandler = (e: any): void => {
```

Comment:
what about when e.data === 'ImmersiveReader-Exit'?

Reply:
That message is already handled, at the bottom of the message handling block.

Reply:
Ah you're right, sorry. The GitHub diff tool isn't the most intuitive :)